improver/utilities/cube_extraction.py

```diff
@@ -250,3 +252,116 @@ def extract_subcube(cube, constraints, units=None, use_original_units=True):
         cube, constraints, units=units, use_original_units=use_original_units
     )
     return output_cube
+
+
+def _create_cutout(cube, grid_spec):
+    """Given a gridded data cube and boundary limits for cutout dimensions,
+    create cutout.  Expects cube on either lat-lon or equal area grid.
+    """
+    x_coord = cube.coord(axis="x").name()
+    y_coord = cube.coord(axis="y").name()
+
+    xmin = grid_spec[x_coord]["min"]
+    xmax = grid_spec[x_coord]["max"]
+    ymin = grid_spec[y_coord]["min"]
+    ymax = grid_spec[y_coord]["max"]
+
+    # need to use cube intersection for circular coordinates (longitude)
+    if x_coord == "longitude":
+        lat_constraint = Constraint(latitude=lambda x: ymin <= x.point <= ymax)
+        cutout = cube.extract(lat_constraint)
+        if cutout is None:
+            return cutout
+
+        cutout = cutout.intersection(longitude=(xmin, xmax))
+
+        # intersection creates a new coordinate with default datatype - we
+        # therefore need to re-cast to meet the IMPROVER standard
+        cutout.coord("longitude").points = cutout.coord("longitude").points.astype(
+            FLOAT_DTYPE
+        )
+        if cutout.coord("longitude").bounds is not None:
+            cutout.coord("longitude").bounds = cutout.coord("longitude").bounds.astype(
+                FLOAT_DTYPE
+            )
+
+    else:
+        x_constraint = Constraint(
+            projection_x_coordinate=lambda x: xmin <= x.point <= xmax
+        )
+        y_constraint = Constraint(
+            projection_y_coordinate=lambda x: ymin <= x.point <= ymax
+        )
+        cutout = cube.extract(x_constraint & y_constraint)
+
+    return cutout
+
+
+def subset_data(cube, grid_spec=None, site_list=None):
+    """Extract a spatial cutout or subset of sites from data
+    to generate suite reference outputs.
+
+    Args:
+        cube (iris.cube.Cube):
+            Input dataset
+        grid_spec (dict):
+            Dictionary containing bounding grid points and an integer "thinning
+            factor" for each of UK and global grid, to create cutouts.  Eg a
+            "thinning factor" of 10 would mean every 10th point being taken for
+            the cutout.
+        site_list (list of str):
```

Comment:
This could also use the `mypy` syntax e.g. `Optional[List[str]]`.

Reply:
Thanks - I think actually I am wrong about this, as you pointed out the WMO site IDs are not always strings.  So I've changed this to be just `list`.